### ISSUE TITLE
`torch.all` the result tensor of Conv2d triggers INTERNAL ASSERT FAILED!

### ISSUE THREAD

**Issue by ilovepytorch** [pytorch/pytorch#70182]
### 🐛 Describe the bug

When doing `torch.all` with the result tensor of `Conv2d` as the input, it will trigger INTERNAL ASSERT FAILED.

```python
import torch
in_channels = 4
out_channels = 1
kernel_size = 1
conv2d = torch.nn.Conv2d(in_channels, out_channels, kernel_size)
input_tensor = torch.rand([4, 4, 2, 2], dtype=torch.float32)
result = conv2d(input_tensor)
torch.all(result)
# RuntimeError: isDifferentiableType(variable.scalar_type())INTERNAL ASSERT FAILED at "/home/mist/pytorch/torch/csrc/autograd/functions/utils.h":65, please report a bug to PyTorch. 
```

### Versions

- PyTorch version: 1.9.0 and 1.10.0

cc @ezyang @gchanan @zou3519 @bdhirsh

**Comment by bdhirsh** [comment]
Marking hi-pri for the internal assert failure.

**Comment by ngimel** [comment]
Cannot repro on 1.10 pip wheel

**Comment by ilovepytorch** [comment]
@ngimel I can repro by using `pip` to install `torch==1.10.1` on MacOs and Ubuntu

**Comment by ngimel** [comment]
Fixed in #65714

**Comment by dagitses** [comment]
Please reopen if not fixed.
